File: Xenon/Xenon/DustEffect.ts
import gsCVector = require("Vector");
import Particle = require("Particle");
import CParticleEffect = require("ParticleEffect");

class CDustEffect extends CParticleEffect {

    constructor() {
        super();
        this.Parent = this;
        this.m_name = "DustEffect";
    }

    //-------------------------------------------------------------

    public createParticle() {
        if (this.m_timer.getTime() > 0.05) {
            this.m_timer.start();

            var p: Particle = new Particle();

            p.m_position = this.getPosition();
            //this.p.m_velocity = new gsCVector::polar(1.0, m_random.getFloat(360.0)) + gsCVector(0.0, 1.0);
            p.m_velocity = new gsCVector(1.0, Math.floor(Math.random() * 360.0)).plus1(new gsCVector(0.0, 1.0));
            p.m_age = 0.0;
            p.m_lifetime = 1.0;
            p.m_mass = 1.0;

            return p;
        }
        else
            return null;
    }

    //-------------------------------------------------------------
}

export = CDustEffect;

File: Xenon-Original_C++_Code/xenon/source/dusteffect.cpp
//-------------------------------------------------------------
//
// Class:	CDustEffect
//
// Author:	John M Phillips
//
// Started:	06/05/00
//
// Base:	CParticleEffect
//
// Derived:	None
//
//-------------------------------------------------------------

#include "game.h"

//-------------------------------------------------------------

Particle *CDustEffect::createParticle()
{
	if (m_timer.getTime() > 0.05f) {
		m_timer.start();

		Particle *p = new Particle;

		p->m_position = getPosition();
		p->m_velocity = gsCVector::polar(1.0f,m_random.getFloat(360.f)) + gsCVector(0.f,1.f);
		p->m_age = 0.f;
		p->m_lifetime = 1.f;
		p->m_mass = 1.f;

		return p;
		}
	else
		return 0;
}

//-------------------------------------------------------------
